account_invoice_pricelist/model/account_invoice_line.py

```diff
@@ -65,6 +69,18 @@ def product_id_change(
                     price_unit, currency, round=True
                 )
                 price_unit = currency.round(price_unit)
+
+        # Fiscal position with Tax Excl / Incl context
+        if fposition_id:
+            fp_tax_ids = res['value']['invoice_line_tax_id']
+            account = AccountAccount.browse(res['value']['account_id'])
+            if type in ('out_invoice', 'out_refund'):
```

Comment:
@legalsylvain here No need to` else`
```
taxes = product.supplier_taxes_id or account.tax_ids
if type in ('out_invoice', 'out_refund'):
    taxes = product.taxes_id or account.tax_ids
```

Reply:
Yes, you're right. But it's not my code, it's a copy of odoo code, so, I'll prefer to keep the same algorigthm, to make the code more easy to compare. 

Are you ok ? 
https://github.com/odoo/odoo/blob/8.0/addons/account/account_invoice.py#L1375

Reply:
@legalsylvain  keep it